rosbag2_transport/include/rosbag2_transport/record_options.hpp

```diff
@@ -24,8 +25,10 @@ struct RecordOptions
 {
 public:
   bool all;
+  bool is_discovery_disabled;
   std::vector<std::string> topics;
   std::string rmw_serialization_format;
+  std::chrono::milliseconds topic_polling_interval;
```

Comment:
would it make sense to expose this to the user as a parameter?

Reply:
This would have a similar expert-level as `--no-discovery`, so yes.